### PR TITLE
Feat/wds color positive

### LINKED ISSUE #23803: feat: WDS semaphore positive color

### PR DESCRIPTION
**Pull request by ichik**
## Description

tl;dr Checking whether seed falls into the interval and picking most distant green hue

Fixes #23803   

## Type of change

- New feature (non-breaking change which adds functionality)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?

- Manual

### Test Plan
Initial POC refinement, no testing necessary

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by KelvinOm — APPROVED** [review]

**Comment by KelvinOm** [comment]
It looks like magic to me. I want to build an interface on all this and see how it will look. I mean to build some kind of complex form with controls and different statuses.
